x/evm/keeper/abci.go

```diff
@@ -22,7 +21,12 @@ func (k *Keeper) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
 	// Gas costs are handled within msg handler so costs should be ignored
 	ctx = ctx.WithGasMeter(sdk.NewInfiniteGasMeter())
 
-	k.SetBlockHash(ctx, req.Header.LastBlockId.GetHash(), req.Header.GetHeight()-1)
+	// Set the hash -> height and height -> hash mapping.
+	hash := req.Header.LastBlockId.GetHash()
+	height := req.Header.GetHeight() - 1
+
+	k.SetHeightHash(ctx, uint64(height), common.BytesToHash(hash))
```

Comment:
why is this no longer in `EndBlock`?

Reply:
because the values won't be exported so there's no need to add it on the end block